### PR TITLE
Step function authoring reset

### LINKED ISSUE #189: Add Reset to StepFunctionAuthoring Component
I would like the option to reset all changes to the original JSON and rendered graph in the `StephFunctionAuthoring` component with the following criteria:

- A "Reset" button is added to the top of the edit form.
- Clicking the button displays a modal with the "Are you sure you want to reset the current Step Function Authoring state?" title and options to "Reset" (destructive) and "Cancel".
- Cancelling or exiting the modal will prevent the reset operation.
- Clicking "Reset" will revert the JSON and rendered graph to it's original state.

### PR DESCRIPTION
**Pull request by jason-ransom-slalom**
This PR is for adding reset functionality to the StepFunctionAuthoring component. If approved, it will close #189 .

Screen Shot 2021-11-10 at 2 42 24 PM

Screen Shot 2021-11-10 at 2 42 40 PM

### CODE REVIEW

**Review by amirkibbar — APPROVED** [review]

File: src/StepFunctionAuthoring/StepFunctionAuthoring.tsx [hunk 1]
```diff
@@ -81,6 +85,11 @@ const StepFunctionAuthoring: FC<StepFunctionAuthoringProps> = ({ initialGraphSta
         // Detect any stored changes
         const [{ change }] = snapshots;
 
+        // Reset state as needed
+        if (change?.type === StephFunctionAuthoringChangeType.RESET) {
```
Comment (adorack-tri) [inline src/StepFunctionAuthoring/StepFunctionAuthoring.tsx:89]: Is there a reason this enum is spelled `StephFunction` instead of `StepFunction`?  Obviously not part of this PR, but if it's just a misspelling we should probably fix it.

**Review by adorack-tri — APPROVED** [review]

**Review by NehalBhandariSlalom — APPROVED** [review]

File: src/StepFunctionAuthoring/StepFunctionAuthoring.tsx (hunk 1 above)
Comment (jason-ransom-slalom) [reply src/StepFunctionAuthoring/StepFunctionAuthoring.tsx:89]: Oh good catch @adorack-tri ! I honestly didn't even realize it (using autocomplete). Will fix.

File: src/StepFunctionAuthoring/StepFunctionAuthoring.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {
 import {
     StephFunctionAuthoringFormState,
```
Comment (adorack-tri) [inline src/StepFunctionAuthoring/StepFunctionAuthoring.tsx:17]: Could you please fix this one too?  Otherwise looks good to me.

Comment (jason-ransom-slalom) [reply src/StepFunctionAuthoring/StepFunctionAuthoring.tsx]: Done. Thanks!
